services/s3/src/main/java/software/amazon/awssdk/services/s3/model/GetUrlRequest.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    /**
+     * @return The key value for this object.
+     */
+    public String key() {
+        return key;
+    }
+
+    /**
+     * @return The region value to use for constructing the URL
+     */
+    public Region region() {
+        return region;
+    }
+
+    /**
+     * @return The endpoint value to use for constructing the URL
+     */
+    public URI endpoint() {
+        return endpoint;
+    }
+
+
+    @Override
+    public Optional<? extends RequestOverrideConfiguration> overrideConfiguration() {
+        return Optional.empty();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new BuilderImpl(this);
+    }
+
+    public static Builder builder() {
+        return new BuilderImpl();
+    }
+
+    public <T> Optional<T> getValueForField(String fieldName, Class<T> clazz) {
+        switch (fieldName) {
+            case "Bucket":
+                return Optional.ofNullable(clazz.cast(bucket()));
+            case "Key":
+                return Optional.ofNullable(clazz.cast(key()));
+            default:
+                return Optional.empty();
+        }
+    }
+
+    @Override
+    public List<SdkField<?>> sdkFields() {
+        return SDK_FIELDS;
+    }
+
+    private static <T> Function<Object, T> getter(Function<GetUrlRequest, T> g) {
+        return obj -> g.apply((GetUrlRequest) obj);
+    }
+
+    private static <T> BiConsumer<Object, T> setter(BiConsumer<Builder, T> s) {
+        return (obj, val) -> s.accept((Builder) obj, val);
+    }
+
+    public interface Builder extends SdkRequest.Builder, SdkPojo, CopyableBuilder<Builder, GetUrlRequest> {
+        /**
+         * Sets the value of the Bucket property for this object.
+         *
+         * @param bucket
+         *        The new value for the Bucket property for this object.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder bucket(String bucket);
+
+        /**
+         * Sets the value of the Key property for this object.
+         *
+         * @param key
+         *        The new value for the Key property for this object.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder key(String key);
+
+        /**
+         * Sets the region to use for constructing the URL.
+         *
+         * @param region
+         *        The region to use for constructing the URL.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder region(Region region);
+
+        /**
+         * Sets the endpoint to use for constructing the URL.
+         *
+         * @param endpoint
+         *        The endpoint to use for constructing the URL.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder endpoint(URI endpoint);
```

Comment:
Should this be endpointOverride to match the semantics of the client builder's builder.endpointOverride?

Reply:
I thought of that but didn't understand why we went with `endpointOverride`. For me, `endpoint` seems better and non-confusing.

Reply:
`endpointOverride` was used because it overrides the endpoint that would be generated from the region with a custom endpoint.